New links in transitions and jump to next transition

Alkarex opened the pull request:
Easier to explain graphically:

<img width="408" height="266" alt="image" src="https://github.com/user-attachments/assets/0e3724a1-155b-4a87-89b3-cfe8a18cb100" />

The jump to next section ⏭ works when the sorting criterion is a date.

Edit: https://github.com/FreshRSS/FreshRSS/pull/8293 is actually needed after all

Alkarex committed:
Add functions to modify a search expression
Needed for upcoming PR.
In particular, I had to implement a proper `__toString()` instead of just returning the raw input string.

Alkarex committed:
Fix PHPStan

Alkarex committed:
Change enforce() to return a clone

Alkarex committed:
New links in transitions and jump to next transition

Alkarex committed:
Remove unused enforce() function

Frenzie commented:
Maybe make it something like 0.8 opacity?

Alkarex committed:
Merge branch 'edge' into transition-links

Alkarex committed:
opacity: 0.5

Alkarex commented:
> Maybe make it something like 0.8 opacity?

I went for `opacity: 0.5`

<img width="412" height="101" alt="image" src="https://github.com/user-attachments/assets/7e867882-6ad7-4877-9f0b-5ad29f9a57b9" />

Alkarex committed:
margin-left: .5em

Alkarex committed:
Revert "Remove unused enforce() function"

This reverts commit ef17d48c7ef1258e52775ef9ae29e4d051a418ec.

Alkarex committed:
Merge branch 'modify-search' into transition-links

Alkarex committed:
Revert to using enforce() function
https://github.com/FreshRSS/FreshRSS/pull/8293

Frenzie reviewed (approved):
Seems like a lot of code but if it's something you like. ;-)

Alkarex committed:
Fix enforce()

Alkarex committed:
Merge branch 'modify-search' into transition-links

Alkarex committed:
Fix hasSameOperators() for pairs of min/max dates

Alkarex committed:
Merge branch 'modify-search' into transition-links

Alkarex commented:
> Seems like a lot of code but if it's something you like. ;-)

It should though be noted that that major part is https://github.com/FreshRSS/FreshRSS/pull/8293 (with general improvements to work with searches), so this PR will actually be quite small.

Alkarex committed:
Merge branch 'edge' into modify-search

Alkarex committed:
Merge branch 'modify-search' into transition-links

Alkarex committed:
Shorter URLs

Alkarex committed:
Use modifiedCurrentRequest

Alkarex committed:
Merge branch 'edge' into transition-links

Alkarex committed:
Merge branch 'edge' into modify-search

Alkarex committed:
More work on parentheses. A bit more to do.

Alkarex committed:
Merge branch 'modify-search' into transition-links

Alkarex committed:
Clean search when changing sorting criteria

Alkarex committed:
Improve parentheses, fix cloning

Alkarex committed:
Merge branch 'edge' into modify-search

Alkarex committed:
Use proper Search toString() in UI

Alkarex committed:
Merge branch 'modify-search' into transition-links

Alkarex committed:
Merge branch 'edge' into transition-links

Alkarex committed:
Remove URL cleaning

Alkarex committed:
Merge branch 'edge' into transition-links

Alkarex committed:
Fix search encoding and quoting
Revised the encoding approach for searches: the HTML encoding is done just before its use for DB search.
Fix also some cases with wrong quoting.
Fix https://github.com/FreshRSS/FreshRSS/pull/8306#issuecomment-3643865439
Follow-up of https://github.com/FreshRSS/FreshRSS/pull/8293

Alkarex committed:
Merge branch 'fix-search-encoding' into transition-links

Alkarex committed:
Merge branch 'edge' into transition-links

Kamaris521 commented:
Is there an option to hide / suppress those transition and jump to headers?

Alkarex commented:
> Is there an option to hide / suppress those transition and jump to headers?

@Kamaris521 See:
* https://github.com/FreshRSS/FreshRSS/discussions/8355#discussioncomment-15346265
* https://github.com/FreshRSS/FreshRSS/discussions/8358#discussioncomment-15353373
